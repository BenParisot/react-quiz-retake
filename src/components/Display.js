import React from 'react';
import PropTypes from 'prop-types';

function Display(headerText) {

  return (
    <section>
      <p>{headerText.headerText}</p>
      <img src={headerText.imageSrc} />
      <p>{headerText.footerText}</p>
    </section>
  );
}
Display.propTypes = {
  headerText: PropTypes.string.isRequired,
  footerText: PropTypes.string.isRequired,
  imageSrc: PropTypes.string.isRequired
};

export default Display;
