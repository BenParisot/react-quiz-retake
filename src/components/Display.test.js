import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('displays the meme', () => {
    const headerText = {
      headerText: 'hello',
      footerText: 'good friend',
      imageSrc: 'blah'
    };
    const wrapper = shallow(<Display headerText={headerText.headerText} footerText={headerText.footerText} imageSrc={headerText.imageSrc} />);
    expect(wrapper).toMatchSnapshot(); 
  });
});
