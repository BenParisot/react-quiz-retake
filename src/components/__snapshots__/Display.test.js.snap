// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component displays the meme 1`] = `
<section>
  <p>
    hello
  </p>
  <img
    src="blah"
  />
  <p>
    good friend
  </p>
</section>
`;
